docs/reference/mapping/runtime.asciidoc

```diff
@@ -765,21 +847,132 @@ and determine which fields to index.
     "hits" : [
       {
         "_index" : "my-index",
-        "_id" : "oWs5KXYB-XyJbifr9mrz",
+        "_id" : "hdEhyncBRSB6iD-PoBqe",
         "_score" : 1.0,
         "_source" : {
-          "@timestamp" : "2020-06-21T15:00:01-05:00",
-          "message" : "211.11.9.0 - - [2020-06-21T15:00:01-05:00] \"GET /english/index.html HTTP/1.0\" 304 0"
+          "timestamp" : "2020-04-30T14:31:27-05:00",
+          "message" : "252.0.0.0 - - [30/Apr/2020:14:31:27 -0500] \"GET /images/hm_bg.jpg HTTP/1.0\" 200 24736"
+        }
+      },
+      {
+        "_index" : "my-index",
+        "_id" : "htEhyncBRSB6iD-PoBqe",
+        "_score" : 1.0,
+        "_source" : {
+          "timestamp" : "2020-04-30T14:31:28-05:00",
+          "message" : "not a valid apache log"
+        }
+      }
+    ]
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took" : $body.took,"timed_out" : $body.timed_out,"_shards" : $body._shards,/]
+// TESTRESPONSE[s/"_id" : "hdEhyncBRSB6iD-PoBqe"/"_id": $body.hits.hits.0._id/]
+// TESTRESPONSE[s/"_id" : "htEhyncBRSB6iD-PoBqe"/"_id": $body.hits.hits.1._id/]
+
+[[runtime-examples-dissect]]
+==== Define a runtime field with a dissect pattern
+<<dissect-processor,Dissect patterns>> are similar to grok patterns, but they
+don't accept regular expressions. This difference allows dissect’s syntax to be
+simple and for some cases faster than grok.
+
+You can use dissect to achieve the same results as a grok pattern, such as the
+<<runtime-examples-grok,`clientip` grok pattern>>. Instead of matching on a log
+pattern, you include the parts of the string that you want to discard. Paying
+special attention to the parts of the string you want to discard will help build
+successful dissect patterns.
+
+[source,console]
+----
+PUT my-index/_mappings
+{
+  "runtime": {
+    "http.client.ip": {
+      "type": "ip",
+      "script": """
+        String clientip=dissect('%{clientip} %{ident} %{auth} [%{@timestamp}] "%{verb} %{request} HTTP/%{httpversion}" %{status} %{size}').extract(doc["message"].value)?.clientip;
+        if (clientip != null) emit(clientip);
+      """
+    }
+  }
+}
+----
+// TEST[continued]
+
+Similarly, you can define a dissect pattern to discard everything in the log
```

Comment:
You aren't discarding things - you are "extracting" the response code. You could even link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status if you are feeling spicy.

Reply:
🔥 🌶️